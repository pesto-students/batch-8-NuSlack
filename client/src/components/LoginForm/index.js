import {
  Form, Icon, Input, Button, Checkbox,
} from 'antd';
import PropTypes from 'prop-types';
import React, { useRef } from 'react';
import { Link } from 'react-router-dom';
import StyledWrapper from './style';

const LoginForm = (props) => {
  const handleSubmit = useRef((e) => {
    e.preventDefault();
    props.form.validateFields((err, values) => {
      if (!err) {
        props.handleFormSubmit(values);
      }
    });
  });

  const { getFieldDecorator } = props.form;
  return (
    <StyledWrapper>
      <Form onSubmit={handleSubmit.current} className="login-form">
        <Form.Item>
          {getFieldDecorator('email', {
            rules: [{ required: true, message: 'Please input your email!' }],
          })(
            <Input
              prefix={<Icon type="user" style={{ color: 'rgba(0,0,0,.25)' }} />}
              placeholder="you@somewhere.com"
            />,
          )}
        </Form.Item>
        <Form.Item>
          {getFieldDecorator('password', {
            rules: [{ required: true, message: 'Please input your Password!' }],
          })(
            <Input
              prefix={<Icon type="lock" style={{ color: 'rgba(0,0,0,.25)' }} />}
              type="password"
              placeholder="Password"
            />,
          )}
        </Form.Item>
        <Form.Item>
          {getFieldDecorator('remember', {
            valuePropName: 'checked',
            initialValue: false,
          })(<Checkbox>Remember me</Checkbox>)}
          <a className="login-form-forgot" href="./forgot-password">
            Forgot password
          </a>
          <Button type="primary" htmlType="submit" className="login-form-button">
            Log in
          </Button>
          Or <Link to="/signup">register now!</Link>
        </Form.Item>
      </Form>
    </StyledWrapper>
  );
};

LoginForm.propTypes = {
  form: PropTypes.shape().isRequired,
  handleFormSubmit: PropTypes.func.isRequired,
};

const WrappedLoginForm = Form.create({ name: 'app_login_form' })(LoginForm);

export default WrappedLoginForm;
