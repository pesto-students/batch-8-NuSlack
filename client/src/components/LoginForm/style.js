import styled from 'styled-components';

const StyledWrapper = styled.div`
  .login-form {
    max-width: 300px;
  }
  .login-form-forgot {
    float: right;
  }
  .login-form-button {
    width: 100%;
    background-color: #008CBA;
  }
`;

export default StyledWrapper;
