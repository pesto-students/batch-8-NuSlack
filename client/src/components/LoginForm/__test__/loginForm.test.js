import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import LoginForm from '../index';

describe('<LoginForm />', () => {
  it('should match the snapshot', () => {
    const tree = shallow(<LoginForm />);
    expect(toJson(tree)).toMatchSnapshot();
  });
});
