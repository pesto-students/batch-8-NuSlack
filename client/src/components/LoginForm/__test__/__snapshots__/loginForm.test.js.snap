// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginForm /> should match the snapshot 1`] = `
<LoginForm
  form={
    Object {
      "getFieldDecorator": [Function],
      "getFieldError": [Function],
      "getFieldInstance": [Function],
      "getFieldProps": [Function],
      "getFieldValue": [Function],
      "getFieldsError": [Function],
      "getFieldsValue": [Function],
      "isFieldTouched": [Function],
      "isFieldValidating": [Function],
      "isFieldsTouched": [Function],
      "isFieldsValidating": [Function],
      "isSubmitting": [Function],
      "resetFields": [Function],
      "setFields": [Function],
      "setFieldsInitialValue": [Function],
      "setFieldsValue": [Function],
      "submit": [Function],
      "validateFields": [Function],
      "validateFieldsAndScroll": [Function],
    }
  }
/>
`;
